src/singleton.h

```diff
@@ -0,0 +1,27 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef SINGLETON_H
+#define SINGLETON_H
+
+#include <Qobject>
+
+template <typename T>
+class Singleton {
```

Comment:
do you use it?

Reply:
No.  I need to pull it.  QObject does not behave well with templates.